### DISCUSSION TITLE
Creating my first plug-in

### DISCUSSION THREAD

**Discussion by ErikLarsson19** [gama-platform/gama#43] [Q&A] [answered]
Im trying to create my first GAMA plug-in in Eclipse. I have installed the GAMA git version och followed the steps off the tutorial on how to create a plug-in from the GAMA documentation. After creating the plug-in project on Eclipse i cannot find the "msi.gama.core" and "msi.gama.ext" in the Dependencies tab when i press "Add". Have anyone had any similar troubles or have found another guide on how to create a plug-in for GAMA i would appreciate it a lot! 

Is there any other way for me to integrate Java code into GAMA? 

Thanks a lot!

**Comment by lesquoyb** [comment]
Hello, maybe @LucasGrjs can help you with that.

**Comment by LucasGrjs** [comment, marked as answer]
Hi Erik, 

Are you on the 2024-06 branch or the 1.9.3 one?
If you are on the 2024-06 branch you should be able to add the plugin "gama.core" and "gama.dependencies" instead of "msi.gama.core" and "msi.gama.ext".
If you are on the 1.9.3 branch. You can try to clean the project in the "Project" tab -> Clean.. Also try to close and reopen Eclipse. If none of this works, try deleting your plugin and create it again following the tutorial.

We did quite a lot of changes on the 2024-06 branch and the documentation on the GAMA website has not been updated accordingly for the "Developing GAMA" section. 

Don't hesitate to ask more questions if you have any trouble on this matter!

**Comment by ErikLarsson19** [comment]
Thank you very much!
This solved my problems so far!

Skickades från E-post<https://go.microsoft.com/fwlink/?LinkId=550986> för Windows

________________________________
Från: Lucas Grosjean ***@***.***>
Skickat: Wednesday, February 28, 2024 5:39:01 AM
Till: gama-platform/gama ***@***.***>
Kopia: Erik ***@***.***>; Author ***@***.***>
Ämne: Re: [gama-platform/gama] Creating my first plug-in (Discussion #43)


Hi Erik,

Are you on the 2024-06 branch or the 1.9.3 one?
If you are on the 2024-06 branch you should be able to add the plugin "gama.core" and "gama.dependencies" instead of "msi.gama.core" and "msi.gama.ext".

We did quite a lot of changes on the 2024-06 branch and the documentation on the GAMA website has not been updated accordingly for the "Developing GAMA" section.

Don't hesitate to ask more questions if you have any trouble on this matter!
